components/calendar/src/iso.rs

```diff
@@ -458,18 +471,38 @@ impl Iso {
     // Lisp code reference: https://github.com/EdReingold/calendar-code2/blob/1ee51ecfaae6f856b0d7de3e36e9042100b4f424/calendar.l#L1237-L1258
     pub(crate) fn iso_from_fixed(date: i32) -> Date<Iso> {
         let year = Self::iso_year_from_fixed(date);
-        let prior_days = date - Self::iso_new_year(year);
+        // Gets the next possible ISO year
+        let next_year = year.saturating_add(1);
+        // Calculates the fixed days difference between the next year and the current year
+        let year_length = Self::fixed_from_iso_integers(next_year, 12, 31)
```

Comment:
Nit: Now that we know this is 365, we should hard-code 365, and add a comment like:

```rust
// Note: The min year, -5879610, is not a leap year, and neither are either of the adjacent years,
// so we can add 365 to keep these calculations in range of an i32.
```